forum/qa-plugin/async-notifications/async-notifications-page.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+class async_notifications_page
+{
+    public function match_request($request)
+    {
+        return ($request === 'async-notifications');
+    }
+
+    public function process_request($request)
+    {
+        // code from https://github.com/q2apro/q2apro-on-site-notifications/blob/master/q2apro-onsitenotifications-layer.php#L107
+        $userid = qa_get_logged_in_userid();
+        if(qa_opt('q2apro_onsitenotifications_enabled') && $userid) {
+            $last_visit = qa_db_read_one_value(
+                qa_db_query_sub(
+                    'SELECT UNIX_TIMESTAMP(meta_value) FROM ^usermeta WHERE user_id=# AND meta_key=$',
+                    $userid, 'visited_profile'
+                ),
+                true
+            );
+            if (is_null($last_visit)) {
+                $last_visit = '1981-03-31 06:25:00';
```

Comment:
This code is copied from https://github.com/q2apro/q2apro-on-site-notifications/blob/master/q2apro-onsitenotifications-layer.php#L107, I didn't modified this.